Update tool lightcurve-analysis to 0.0.1+galaxy0

dsavchenko opened the pull request.

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/3e4a5accbebb88ee0ff6987819504ae9aba940c4

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/bed634c3e276ae9bc7f89c0a66b4d1d812b96db6

volodymyrss commented:
With first commit the default was used instead of input.

With second commit the tool disappeared :(

https://galaxy.odahub.fr/

dsavchenko commented:
> With second commit the tool disappeared :(

This is some transient problem which I already saw before. The last installed tool may randomly disappear. 

Probably, Galaxy requires restarting upon adding the tool (at least, this seemed to help). Currently, I'm injecting tools to the live instance. 
I will add the restart to the playbook. It may introduce some minor downtime, but it's not critical here. We can explore how tool registering is happening in Galaxy in more details in the future.

dsavchenko commented:
> With first commit the default was used instead of input.

This is still the case. I'm having a look

volodymyrss commented:
> > With second commit the tool disappeared :(
> 
> This is some transient problem which I already saw before. The last installed tool may randomly disappear.
> 
> Probably, Galaxy requires restarting upon adding the tool (at least, this seemed to help). Currently, I'm injecting tools to the live instance. I will add the restart to the playbook. It may introduce some minor downtime, but it's not critical here. We can explore how tool registering is happening in Galaxy in more details in the future.

Ok thanks, the tool is back.

The tool fails somehow, looks like the parameter is not substituted https://galaxy.odahub.fr/datasets/98e8737e708a5bb0/details

dsavchenko committed:
empty

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/06e0de19b80bacc2cc68e5b9e0af5a9a23a30ead

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/29d29c7eaac2ea309e9829b58e9068af4e81ecae

dsavchenko commented:
> The tool fails somehow, looks like the parameter is not substituted

Fixed

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/eac8b065c2ca1e824b1e01a7341d30d2297598f6

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/a20896ba5edba3d5e9c63506e0bf867225dcf3b9

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/7405a3315b38fb70433802c049cb0c37cfd941bc

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/486d4bb34d958eb876c2537baf77a671cf11d9e6

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/c47b095499372ce5177f9cc8f25c1a8f943252a6

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/e91425e3218e47df7f7b31543d8bb10e0bcca6e7

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/5094132dd5e5f577ade4e843492603435e0a7656

volodymyrss commented:
This works pretty well!

https://galaxy.odahub.fr/u/volodymyr/w/grb-analysis-workflow

Here I normalize and then combine HESS and SPI-ACS data. It's a of a mock-up of course, I should have started from image or spectra. Or maybe I refine the LC if I find some common variable source, probably the BOAT GRB.

![image](https://github.com/esg-epfl-apc/tools-astro/assets/3909535/e90f7e41-f2cb-4102-8e25-8cd580126657)

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/5094132dd5e5f577ade4e843492603435e0a7656

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/5094132dd5e5f577ade4e843492603435e0a7656

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/61f02819fcdaeaa43ec92db0949db7a029fd71c9

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/c8bab344dc771c571e7e347f34ecf20c53543c7b

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/79ac0083817eb87ef0be9e71ac7a77df5dc7214a

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/494ed7bc970180eb3f06719a5ca64e6ef821edd2

dsavchenko committed:
Merge branch 'main' into auto-update-galaxy-tool-lightcurve-analysis-v0.0.1-galaxy0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/fdeef1d7e716c836601f41dc961ebfb288fa456f

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/fdeef1d7e716c836601f41dc961ebfb288fa456f

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/fdeef1d7e716c836601f41dc961ebfb288fa456f

dsavchenko committed:
Merge branch 'main' into auto-update-galaxy-tool-lightcurve-analysis-v0.0.1-galaxy0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/fdeef1d7e716c836601f41dc961ebfb288fa456f

dsavchenko committed:
Merge branch 'main' into auto-update-galaxy-tool-lightcurve-analysis-v0.0.1-galaxy0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/349f535a534a8abe8d592e116b1113721af6dd24

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/lightcurve-analysis/-/commit/d1760f8c71d7a462cba7b8f01e0de62c0bc4345c

dsavchenko commented:
@volodymyrss 
This passes tests but misses help and citations. Do you have time this week to finalize? 
It's used in https://workflowhub.eu/workflows/766, and it would be great to polish the tools so that we can mark the workflow stable, and it will be able to run on usegalaxy.eu (the grb detection tool may be more difficult though, I will comment there)

dsavchenko commented:
After looking into notebooks, this seems to be really a mock-up version of the tool. 

The mode is only used in the "detect" notebook and the only value that matters is "Background". 
The peak detection is a simple max, does it make much sense? 
The "correlate" one does nothing but plotting both lightcurves (and no peak detection there at all).

I'm not sure, I can quickly improve the tool, not being very skilled in time-domain. Should we leave the workflow as WIP @volodymyrss ?

volodymyrss commented:
Lets leave it maybe. Ok could you remind me Friday if I do not do it?
